pyvista/__init__.py

```diff
@@ -74,3 +54,46 @@
 
 # Name used for unnamed scalars
 DEFAULT_SCALARS_NAME = 'Data'
+
+MAX_N_COLOR_BARS = 10
+
+
+# Lazily import/access the plotting module
+def __getattr__(name):
+    """Fetch an attribute ``name`` from ``globals()`` or the ``pyvista.plotting`` module..
+
+    This override is implemented to prevent importing all of the plotting module
+    and GL-dependent VTK modules when importing PyVista.
+
+    Raises
+    ------
+    AttributeError
+        If the attribute is not found.
+
+    """
+    try:
+        return globals()[name]
+    except KeyError:
+        pass
+
+    import importlib
+    import inspect
+
+    allow = {
+        'demos',
+        'examples',
+        'ext',
+        'trame',
+        'utilities',
+    }
+    if name in allow:
+        return importlib.import_module(f'pyvista.{name}')
```

Comment:
Couldn't we just explicitly import these subpackages rather than deferring them to `__getattr__()`? They don't look very plotty to me.

Reply:
They may not look "plotty" but they do require the plotting module. Why?:

- `examples`: Use of `pyvista.Texture`
- `demos`: functions like `plot_glyphs()`
- `ext`: Eh, this could be fine since no modules are imported in the `__init__.py` but the modules each require te plotting module
- `trame`: requires the plotting module (mainly for type annotations)
- `utilities`: this is deprecated but can point to the plotting module if accessed

Reply:
The other part. These were not previously imported at the top-level namespace. The usage is always: `from pyvista.examples import ...`, never import pyvista; pyvista.examples...`

Reply:
I looked into one of the modules when I posted this comment and it seemed to me that we don't need plotting to _import_ the modules. Which concurs with your summary (thanks for that!). So do I understand correctly that you're mostly motivated by the future plan to separate the plotting functionality into a package, which is why you also "axe" functionality that needs GL at runtime?

Reply:
> mostly motivated by the future plan to separate the plotting functionality into a package

Correct -- this may not happen now or in the next few months but it's important to have this level of decoupling from now on so that we have the option/ability to ship as separate packages down the road.